opentracing/span.py

```diff
@@ -63,18 +62,17 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                                             'tb': exc_tb})
         self.finish()
 
-    def start_child(self, operation_name):
+    def start_child(self, operation_name, tags=None):
```

Comment:
I think this might be the case where language differences may result in minor divergence in the APIs. Accepting optional tags would be ugly in Go, but elegant in Python since the arg is not required. There are also performance reasons to prefer the second form when available - one less function call, one less mutex lock, and possibly one less memory allocation for the dictionary. 

We actually haven't talked about memory allocation, I had a comment somewhere that by passing a dictionary to some call, the caller gives up ownership of that dictionary. We can either emphasize safety by requiring tracer to always take copies of passed collections if it wants to store them, or emphasize performance by telling the caller that it must always give up ownership of collections.